pkg/remote/snapshot.go

```
@@ -0,0 +1,485 @@
+
+		destination := filepath.Join(root, filepath.FromSlash(rel))
+		if !pathInsideRoot(root, destination) {
+			return fmt.Errorf("repository archive path %q escapes destination", header.Name)
+		}
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.MkdirAll(destination, 0o755); err != nil {
+				return fmt.Errorf("failed to create archive directory %q: %w", rel, err)
+			}
+		case tar.TypeReg, tar.TypeRegA:
```

golangci-lint (2.12.2)

[error] SA1019: tar.TypeRegA has been deprecated since Go 1.11 and an alternative has been available since Go 1.1: Use TypeReg instead. 

(staticcheck)